openmetadata-service/src/main/java/org/openmetadata/service/security/auth/BasicAuthenticator.java

```diff
@@ -0,0 +1,477 @@
+    tokenRepository.deleteTokenByUserAndType(user.getId().toString(), PASSWORD_RESET.toString());
+    tokenRepository.insertToken(resetToken);
+  }
+
+  @Override
+  public void resetUserPasswordWithToken(UriInfo uriInfo, PasswordResetRequest request) throws IOException {
+    String tokenID = request.getToken();
+    PasswordResetToken passwordResetToken = (PasswordResetToken) tokenRepository.findByToken(tokenID);
+    if (passwordResetToken == null) {
+      throw new EntityNotFoundException("Invalid Password Request. Please issue a new request.");
+    }
+    List<String> fields = userRepository.getAllowedFieldsCopy();
+    fields.add(USER_PROTECTED_FIELDS);
+    User storedUser =
+        userRepository.getByName(
+            uriInfo, request.getUsername(), new EntityUtil.Fields(fields, String.join(",", fields)));
+    // token validity
+    if (!passwordResetToken.getUserId().equals(storedUser.getId())) {
+      throw new RuntimeException("Token does not belong to the user.");
+    }
+    verifyPasswordResetTokenExpiry(passwordResetToken);
+    // passwords validity
+    if (!request.getPassword().equals(request.getConfirmPassword())) {
+      throw new RuntimeException("Password and Confirm Password should match");
+    }
+    PasswordUtil.validatePassword(request.getPassword());
+
+    String newHashedPwd = BCrypt.withDefaults().hashToString(12, request.getPassword().toCharArray());
+    BasicAuthMechanism newAuthForUser = new BasicAuthMechanism().withPassword(newHashedPwd);
+
+    storedUser.setAuthenticationMechanism(new AuthenticationMechanism().withAuthType(BASIC).withConfig(newAuthForUser));
+
+    userRepository.createOrUpdate(uriInfo, storedUser);
+
+    // delete the user's all password reset token as well , since already updated
+    tokenRepository.deleteTokenByUserAndType(storedUser.getId().toString(), PASSWORD_RESET.toString());
+
+    // Update user about Password Change
+    try {
+      EmailUtil.getInstance().sendAccountStatus(storedUser, "Update Password", "Change Successful");
+    } catch (Exception ex) {
+      LOG.error("Error in sending Password Change Mail to User. Reason : " + ex.getMessage());
+      throw new CustomExceptionMessage(424, EMAIL_SENDING_ISSUE);
+    }
+    loginAttemptCache.recordSuccessfulLogin(request.getUsername());
+  }
+
+  @Override
+  public void changeUserPwdWithOldPwd(UriInfo uriInfo, String userName, ChangePasswordRequest request)
+      throws IOException {
+    // passwords validity
+    if (!request.getNewPassword().equals(request.getConfirmPassword())) {
+      throw new RuntimeException("Password and Confirm Password should match");
+    }
+    PasswordUtil.validatePassword(request.getNewPassword());
+    List<String> fields = userRepository.getAllowedFieldsCopy();
+    fields.add(USER_PROTECTED_FIELDS);
+
+    // Fetch user
+    User storedUser =
+        userRepository.getByName(uriInfo, userName, new EntityUtil.Fields(fields, String.join(",", fields)));
```

Comment:
any reason not to query using "*" for fields here

Reply:
we are updating the user's password and use the method createOrUpdate , if i don't get all the fields and for example just use "authenticationMechanism" to get User , the missing fields for example isEmailVerified which i did not fetch are all updated to the current value that is null in this case, so i need to fetch all the fields for intenal calls , so that update does not cause changes to other fields